.dotnet.azure/sdk/openai/Azure.AI.OpenAI/tests/FineTuningTests.cs

```diff
@@ -332,29 +329,17 @@ private async Task<OpenAIFileInfo> UploadAndWaitForCompleteOrFail(FileClient fil
         return uploadedFile;
     }
 
-    private Task<FineTuningJob> WaitForJobToEnd(FineTuningClient client, FineTuningJob job)
-    {
-        RequestOptions options = new();
-        string jobId = job.ID;
-
-        // NOTE: Fine tuning jobs can take up 30 minutes to complete so the timeouts here are longer to account for that
-        return WaitUntilReturnLast(
-            job,
-            async () =>
-            {
-                ClientResult result = await client.GetJobAsync(jobId, options).ConfigureAwait(false);
-                return ValidateAndParse<FineTuningJob>(result);
-            },
-            j => j.Status == "cancelled" || j.Status == "failed" || j.Status == "succeeded",
-            TimeSpan.FromMinutes(1),
-            TimeSpan.FromMinutes(40));
-    }
-
     private IAsyncEnumerable<FineTuningJob> EnumerateJobsAsync(FineTuningClient client)
         => EnumerateAsync<FineTuningJob>((after, limit, opt) => client.GetJobsAsync(after, limit, opt));
 
     private IAsyncEnumerable<FineTuningCheckpoint> EnumerateCheckpoints(FineTuningClient client, string jobId)
-        => EnumerateAsync<FineTuningCheckpoint>((after, limit, opt) => client.GetJobCheckpointsAsync(jobId, after, limit, opt));
+        => EnumerateAsync<FineTuningCheckpoint>((after, limit, opt) =>
+        {
+            var jobIdBinaryString = $"{{\"jobId\":\"{jobId}\"}}";
+            var createdToken = ContinuationToken.FromBytes(BinaryData.FromString(jobIdBinaryString));
+            CreateJobOperation operation = CreateJobOperation.Rehydrate(client, createdToken);
+            return (IAsyncEnumerable<ClientResult>)operation.GetJobCheckpoints(after, limit, opt);
+        });
 
     private async IAsyncEnumerable<T> EnumerateAsync<T>(Func<string?, int?, RequestOptions, IAsyncEnumerable<ClientResult>> getAsyncEnumerable)
```

Comment:
Just a heads-up that these APIs have changed in SCM 1.1.0 (based on your feedback!), and this will need to be updated as a result of that once the upgrade PR lands.  Feel free to reach out once that happens if you need assistance on that.

Reply:
I'm hoping we can get this PR in soon. It's already WAAY too big :(